### PR TITLE
HZN-1280: Added cache for flow exporters and their interfaces

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/HZN-1280

### CODE REVIEW

File: features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java [hunk 1]
```diff
@@ -557,6 +602,10 @@ public void failed(Exception ex) {
         return future;
     }
 
+    private void markNodeWithFlows(final Integer nodeId, final Integer inputInterfaceId, final Integer outputInterfaceId) {
```
[the remark this answers is not in the document]
Comment (christianpape) [reply features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java]: Ok, I forgot

File: features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/InitializingFlowRepository.java [hunk 2]
```diff
@@ -82,15 +82,15 @@ public void persist(Collection<Flow> flows, FlowSource source) throws FlowExcept
...
-    @Override
-    public CompletableFuture<Set<Integer>> getExportersWithFlows(int limit, List<Filter> filters) {
-        return delegate.getExportersWithFlows(limit, filters);
-    }
-
-    @Override
-    public CompletableFuture<Set<Integer>> getSnmpInterfaceIdsWithFlows(int limit, List<Filter> filters) {
-        return delegate.getSnmpInterfaceIdsWithFlows(limit, filters);
-    }
+//    @Override
```
[the remark this answers is not in the document]
Comment (christianpape) [reply features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/InitializingFlowRepository.java]: Yes

File: features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/FlowRestService.java [hunk 3]
```diff
@@ -70,38 +70,25 @@
...
      * @return node summaries
      */
     @GET
     @Path("exporters")
     @Produces(MediaType.APPLICATION_JSON)
-    List<FlowNodeSummary> getFlowExporters(
-            @DefaultValue(DEFAULT_LIMIT) @QueryParam("limit") final int limit,
-            @Context final UriInfo uriInfo
-    );
+    List<FlowNodeSummary> getFlowExporters();
```
[the remark this answers is not in the document]
Comment (christianpape) [reply features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/FlowRestService.java]: done

**Comment by j-white** [comment]
Can we look at re-targeting this for `release-22.0.1`?

**Comment by fooker** [comment]
Rebased to `release-22.0.1'.

**Pull request merged**